services/src/test/scala/cromwell/pipeline/service/ProjectConfigurationServiceTest.scala

```diff
@@ -39,61 +45,77 @@ class ProjectConfigurationServiceTest extends AsyncWordSpec with Matchers with M
+        when(configurationRepository.getConfigurationById(projectId)).thenReturn(Future.successful(None))
         when(configurationRepository.addConfiguration(activeConfiguration)).thenReturn(Future.successful(result))
-        configurationService.addConfiguration(activeConfiguration).map(_ shouldBe result)
+        configurationService.addConfiguration(activeConfiguration, userId).map(_ shouldBe result)
       }
 
       "return success if update was successful" in {
         val result: Unit = ()
-        when(configurationRepository.getById(projectId)).thenReturn(Future.successful(Some(activeConfiguration)))
+        when(configurationRepository.getConfigurationById(projectId))
+          .thenReturn(Future.successful(Some(activeConfiguration)))
         when(configurationRepository.updateConfiguration(updatedConfiguration)).thenReturn(Future.successful(result))
-        configurationService.addConfiguration(updatedConfiguration).map(_ shouldBe result)
+        configurationService.addConfiguration(updatedConfiguration, userId).map(_ shouldBe result)
       }
 
       "return failure if it couldn't fetch existing configuration" in {
         val error = new Exception("Oh no")
-        when(configurationRepository.getById(projectId)).thenReturn(Future.failed(error))
-        configurationService.addConfiguration(activeConfiguration).failed.map(_ shouldBe error)
+        when(configurationRepository.getConfigurationById(projectId)).thenReturn(Future.failed(error))
+        configurationService.addConfiguration(activeConfiguration, userId).failed.map(_ shouldBe error)
       }
 
       "return failure if creation wasn't successful" in {
         val error = new Exception("Oh no")
-        when(configurationRepository.getById(projectId)).thenReturn(Future.successful(None))
+        when(configurationRepository.getConfigurationById(projectId)).thenReturn(Future.successful(None))
         when(configurationRepository.addConfiguration(activeConfiguration)).thenReturn(Future.failed(error))
-        configurationService.addConfiguration(activeConfiguration).failed.map(_ shouldBe error)
+        configurationService.addConfiguration(activeConfiguration, userId).failed.map(_ shouldBe error)
       }
 
       "return failure if update wasn't successful" in {
         val error = new Exception("Oh no")
-        when(configurationRepository.getById(projectId)).thenReturn(Future.successful(Some(activeConfiguration)))
+        when(configurationRepository.getConfigurationById(projectId))
+          .thenReturn(Future.successful(Some(activeConfiguration)))
         when(configurationRepository.updateConfiguration(activeConfiguration)).thenReturn(Future.failed(error))
-        configurationService.addConfiguration(activeConfiguration).failed.map(_ shouldBe error)
+        configurationService.addConfiguration(activeConfiguration, userId).failed.map(_ shouldBe error)
+      }
+
+      "return failure creation if user is not owner for project" in {
```

Comment:
Maybe change wording a little to something like "return failure if user is not the project's owner"

Reply:
Ok. I will change it.

Reply:
Changed.